app/javascript/packs/custom.js

```diff
@@ -0,0 +1,344 @@
+$(document).ready(function() {
+  if (typeof(Storage) !== "undefined") {
+    var addButtons = $('.btn-adding');
+    var resetButton = $('#btn-reset');
+    var dishNames = $('.item-restaurant-name');
+    var dishPrices = $('.current-price');
+    var totalOrderNumber = $('#total-order-number');
+    var orderCardGroup = $('#now-order-card-group');
+    var totalAmounts = $('.total-amount');
+    var orderNumbers;
+    var billDishNames;
+    var billDishPrices;
+    var plusButtons;
+    var minusButtons;
+
+    checkToStartSession();
+    totalOrderNumber.text(sessionStorage['total-order-number']);
+
+    for (let i = 0; i < sessionStorage['storage-lenght'];i++) {
+      addSessionItem(i);
+      totalAmounts[0].textContent = sessionStorage['total-amount'];
+      totalAmounts[1].textContent = sessionStorage['total-amount'];
+      plusButtons = $('.fa-plus-square');
+      minusButtons = $('.fa-minus-square');
+      let num = plusButtons.length-1;
+      plusButtons[num].addEventListener('click', function(){plusItem(this);});
+      minusButtons[num].addEventListener('click', function(){minusItem(this);});
+    }
+
+    orderNumbers = $('.number-order');
+    billDishNames = $('.name-order');
+    billDishPrices = $('.price-order');
+
+    for (let i = 0; i<addButtons.length; i++) {
+      addButtons[i].addEventListener('click', function(){ addItem(i); });
+    }
+
+    resetButton.on('click', resetOrder);
+
+    function checkToStartSession() {
+      if (sessionStorage.getItem('storage-lenght') === null) {
+        sessionStorage['storage-lenght'] = 0;
+      }
+
+      if (sessionStorage.getItem('total-order-number') === null) {
+        sessionStorage['total-order-number'] = 0;
+      }
+    }
+
+    function resetOrder() {
+      console.log('Click');
+      totalOrderNumber.text(0);
+      resetTotalAmount();
+      sessionStorage.clear();
+      orderCardGroup.empty();
+      billDishNames = $('.name-order');
+    }
+
+    function resetTotalAmount() {
+      totalAmounts[0].textContent = 0;
+      totalAmounts[1].textContent = 0;
+    }
+
+    function increaseTotalNumber() {
+      let number = totalOrderNumber.text();
```

Comment:
khác nhau giữa let và var là gì e nhỉ
ở đây c dùng var dc k

Reply:
Let chỉ có phạm vi trong block chứ nó còn var có phạm vi xuyên suốt function chứa nó.
Vậy ở đây nên dùng var phải ko ạ.

Reply:
Và cả các function tương tự bên dưới nữa ạ

Reply:
thường thì c dùng var e ạ

Reply:
e dùng var đi

Reply:
thử thêm lodash vào xem code của e có ngắn và hay hơn k. nó cũng khá dễ
thay vì for kia e dùng _.forEach của lodash. nó hỗ trợ nhiều function https://lodash.com/docs/4.17.15

Reply:
Vâng em sẽ thử ạ. Em cảm ơn chị

Reply:
Có những hàm em chỉ check điều kiện xong break nên em giữ nguyên